File: frontend/src/services/resourceService.js
import api from "../api/axios.js";

const BASE = "/resources";

const resourceService = {
  getAll:    (params)     => api.get(BASE, { params }),
  getById:   (id)         => api.get(`${BASE}/${id}`),
  create:    (data)       => api.post(BASE, data),
  update:    (id, data)   => api.put(`${BASE}/${id}`, data),
  remove:    (id)         => api.delete(`${BASE}/${id}`),
};

export default resourceService;

File: backend/src/main/java/com/sliit/group66/smartcampus/service/ResourceService.java
package com.example.resource.service;

import com.example.resource.dto.ResourceDTO;
import java.util.List;

public interface ResourceService {

    ResourceDTO create(ResourceDTO dto);

    List<ResourceDTO> getAll();

    ResourceDTO getById(Long id);

    ResourceDTO update(Long id, ResourceDTO dto);

    void delete(Long id);
}